scripts/cplane/cluster_init.sh

```diff
@@ -0,0 +1,351 @@
+    echo "--------------------------------------"
+}
+
+
+# Copy the k8s admin.conf into the user directory for subsequent use
+function kubeconf_copy {
+    if [ -f /etc/kubernetes/admin.conf ] ; then
+        echo "⚙️  Create local '.kube/config'"
+        mkdir -p ${HOME}/.kube
+        sudo cp -f /etc/kubernetes/admin.conf ${HOME}/.kube/config
+        sudo chown $(id -u):$(id -g) ${HOME}/.kube/config
+    fi
+}
+
+
+# Install the Flannel CNI (Container Network Interface)
+function install_flannel_cni {
+    echo "🚜  Install Flannel CNI Service"
+
+    # The Prepackaged Flannel CNI is hard coded to use CIDR of 10.244.0.0/16
+    if ! [ "${POD_BASE_CIDR}" == "10.244.0.0" ]; then
+        echo "❌ For Flannel CNI via 'kubectl apply', var 'POD_BASE_CIDR' must be '10.244.0.0'"
+        echo "   (POD_BASE_CIDR is currently defined as '${POD_BASE_CIDR}')"
+        echo ""
+        exit 1
+    fi
+
+    # Install Flannel Service
+    echo "--------------------------------------"
+    kubectl apply -f https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml --validate=false
+    echo "--------------------------------------"
+
+    # Restart the kubelet
+    sudo service kubelet restart
+}
+
+
+# Verify the Flannel CNI Installation
+function verify_flannel_cni {
+    echo "🔍 Verify Flannel CNI Service Install"
+
+    # Verify that CNI service is running
+    timeout=60 # 1 minute = 60 seconds
+    interval=10 # check every 10 seconds
+    elapsed=0
+
+    # Repeat loop until time runs out
+    while [ ${elapsed} -lt ${timeout} ]; do
+        echo -n "🔍 Verify Flannel Service running"
+        if [ ${elapsed} -gt 0 ]; then
+            echo " (trying for $((timeout - elapsed)) more seconds)"
+        else
+            echo ""
+        fi
+
+        # If Flannel is running correctly then it will report back the pods list
+        flannel_state=$(kubectl get pods -n kube-flannel -o json 2>/dev/null | jq -r '.items[] | select(.metadata.name | startswith("kube-flannel-")) | .status.phase')
+
+        elapsed=$((elapsed + interval))
+        if [ ${flannel_state} == "Running" ]; then
+            break
+        elif [ ${elapsed} -lt ${timeout} ]; then
+            sleep ${interval}
+        fi
+    done
+
+    if [ ${flannel_state} == "Running" ]; then
+        echo "✅ Flannel CNI Service is running"
+    else
+        echo "❌ Timed out during check - Flannel CNI service is NOT running"
+        exit 1
+    fi
+}
+
+
+# Install the Weave CNI (Container Network Interface)
+# NOTE: Weave Project was Shuttered in June 2024 and no longer supported
+function install_weave_cni {
+    echo "🚜  Install Weave CNI Service"
+    echo "--------------------------------------"
+    kubectl apply -f https://github.com/weaveworks/weave/releases/download/v2.8.1/weave-daemonset-k8s.yaml --validate=false
+    echo "--------------------------------------"
+}
+
+
+# Verify the Weave CNI Installation
```

Comment:
the verify happens after the install.  
Automation uses consistent "gating" and checking to ensure expected state before moving to the next step.